server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessAttributeHealth.java

```diff
@@ -0,0 +1,335 @@
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * Cluster Awareness health information
+ *
+ */
+public class ClusterAwarenessAttributeHealth implements Iterable<ClusterAwarenessAttributeValueHealth>, Writeable, ToXContentFragment {
+
+    private final String awarenessAttributeName;
+    private Map<String, ClusterAwarenessAttributeValueHealth> awarenessAttributeValueHealthMap;
+    private final String ACTIVE = "active";
+    private final String INITIALIZING = "initializing";
+    private final String RELOCATING = "relocating";
+
+    /**
+     * Creates Awareness AttributeValue Health information
+     *
+     * @param awarenessAttributeValue Awareness Attribute value ie zone, rack etc
+     * @param weightedRoutingWeights  Weights Map of Awareness Attribute
+     * @param displayUnassignedShardLevelInfo Governs if unassigned info should be visible or not
+     * @param shardAllocationPerNode Shard Allocation per node
+     * @param nodeMap The nodeMap stored in cluster state
+     * @param totalShards Total shards currently in cluster
+     */
+    public ClusterAwarenessAttributeHealth(
+        String awarenessAttributeValue,
+        Map<String, Double> weightedRoutingWeights,
+        boolean displayUnassignedShardLevelInfo,
+        Map<String, Map<String, Integer>> shardAllocationPerNode,
+        ImmutableOpenMap<String, DiscoveryNode> nodeMap,
+        int totalShards
+    ) {
+
+        awarenessAttributeName = awarenessAttributeValue;
+
+        // This is the Map which is storing the per attribute value health stats.
+        // The key would be the attribute value like rack-1 and teh stats would be stored as value.
+        Map<String, ClusterAwarenessAttributeValueHealth> attributesHealth = new HashMap<>();
+
+        // This is the map that would store all the stats per attribute ie
+        // health stats for rack-1, rack-2 etc.
+        awarenessAttributeValueHealthMap = new HashMap<>(Collections.emptyMap());
+
+        ClusterAwarenessAttributeValueHealth clusterAwarenessAttributeValueHealth;
+        String attributeValue;
+
+        if (!nodeMap.isEmpty()) {
+            Iterator<String> iter = nodeMap.keysIt();
+            while (iter.hasNext()) {
+                String node = iter.next();
+                DiscoveryNode nodeDiscovery = nodeMap.get(node);
+                Map<String, String> nodeAttributes = nodeDiscovery.getAttributes();
+                if (!nodeAttributes.isEmpty()) {
+                    if (nodeAttributes.containsKey(awarenessAttributeName)) {
+                        attributeValue = nodeAttributes.get(awarenessAttributeName);
+
+                        if (!attributesHealth.containsKey(attributeValue)) {
```

Comment:
If it doesn't contains key, let initialize with 0 values. Rest of the code could be same after that .

Reply:
It better if we can split the PR into 2 - one having core changes w/o the Transport layer and the rest layer. Then other having rest of the changes . We did similar thing for decommission as well .